### PR TITLE
Update RSE in R learning goals and narrative

### PR DESCRIPTION
**Pull request by lwjohnst86**
Most of the Python outline didn't make sense in an R view, so I've revised the course design as a starting point for how to proceed with the R side. 

Of course, this is ON HOLD until we finish the Python side. I'm only starting this because as I mentioned, I feel I've hit the limit of my capabilities with Python RSE, so might as well start working on this.

~~This is based on the branch that makes up PR #330, so that needs to be merged first before moving on to this one.~~

### CODE REVIEW

File: r-rse/course-design.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    - Deadlines for milestones to acheive for project, with instructor feedback on progress
+    - Out-of-class formative assessments based on maybe learnr tutorials?
+    - After completing in-class exercises, go over them as class?
+
+- Key resources to use for this:
+  - R packages: usethis, devtools, drake?, pkgdown, testthat, styler, rmarkdown
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
```
Comment (cwickham) [inline r-rse/course-design.md:45]: Right, I think of R packages as being a reference book, we are writing for people learning this for the first time.

File: r-rse/course-design.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    - In class portion: 
+        - Some lectures for those material that aren't coding heavily
+        (e.g. for lessons to involve community, being a good community member)
+        - Live coding (with instructor)
+        - Exercises that expand on material covered with instructor that 
+        solve a provided problem, with or without scaffolding
+    - Out of class: 
+        - Group work/project creating an R package based on a set of (given?) problems
+        - Formative assessments like coding with immediate feedback on correct/incorrect
+        (via learnr package? Esp with new update to RStudio in version 1.3)
```
Comment (cwickham) [inline r-rse/course-design.md:32]: This would be awesome! But, writing correctness tests is time consuming...

File: r-rse/course-design.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - What makes this course different
+
+- Folder and file structure (RStudio R Projects)
+    - Filepaths, very basics of the shell (in RStudio)
+    - Making use of the fs package for filesystem management
+    
+- Setting up an R package
```
Comment (cwickham) [inline r-rse/course-design.md:61]: Does this start with what a package is, and why you might want to make one?

File: r-rse/course-design.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    - Using devtools, usethis for development
+        - e.g. `use_devtools()` and `use_usethis()`.
+    - Developing functions
```
Comment (cwickham) [inline r-rse/course-design.md:64]: Can you elaborate? Is this the mechanics of turning a code snippet into a function?  Or is it something more?

File: r-rse/course-design.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    - Making and using datasets
+
+- Function development in a package environment
+    - Dependency management
+    - Function documentation (with roxygen2), part 1
+        - To use `@examples` to help with creating function
+
+- (Idea) Additional features to make use of
+    - RStudio addins, snippets, rmarkdown templates, templates
+    - rstudioapi?
```
Comment (cwickham) [inline r-rse/course-design.md:74]: You mean as possible components of a package, right?  I think of these as things beyond functions that an app can provide. Having a survey of them in one place would be really nice (I can't think of anywhere else that does).   

Would shiny apps fit here too?

File: r-rse/course-design.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+- Version control: Using Git and GitHub as a sole user (part 1)
+    - Using Git in RStudio (standard add-commmit-history)
+    - Using Git in the Terminal of RStudio (moving in history with checkout,
+    creating branches, adding and updating remotes)
+    - Setting up GitHub for R package (make use of usethis), pushing and pulling
+
+- Checking correctness of code
+    - Unit test (with testthat)
+    - Code coverage (with covr)
+    - With devtools
```
Comment (cwickham) [inline r-rse/course-design.md:85]: Are our learners already OK with debugging?  E.g. I'm thinking about Jenny Bryan's [Object of closure is not sub-settable keynote](https://rstudio.com/resources/rstudioconf-2020/object-of-type-closure-is-not-subsettable/), where unit tests are the last strategy she covers, do we assume the learners already know the earlier strategies?

File: r-rse/course-design.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+- Build management
+    - With devtools
+    - Maybe with drake?
```
Comment (cwickham) [inline r-rse/course-design.md:89]: My knowledge of drake is minimal...for some reason I've thought about it more as a tool for automating data pipelines.

File: r-rse/course-design.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    - Labeling
+    - Project boards
+    - Milestones
+
+- Preparing for package release
+    - Check builds on other systems (with rhub)
+    - Running local CRAN checks
+    - Managing your versions, git tagging
+
+- Making your R package development to a team-based environment
```
Comment (cwickham) [inline r-rse/course-design.md:129]:
```suggestion
- R package development in a team-based environment
```

File: r-rse/course-design.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    - Making use of GitHub Organizations/Teams
+    - Branch protection, member roles
+    - Decision making
+    - Governance
+    - Issue assignment
+    - Relying on and using code of conduct to build culture and standards
```
Comment (cwickham) [inline r-rse/course-design.md:135]: It feels like working on an R package with more than one person doesn't get covered until the end of the book, but in a class setting learners might be working together from very early on in the class.  Can we make this a different "part", and we give some guidance to instructors up front on different times pieces might be covered? E.g. you could imagine covering a classroom Code of Conduct on the first day, and get students to read about why it's a good idea to have one from this section, or when students start on their first group project, you might get them to select and agree to a Code of Conduct for their own group.

File: r-rse/course-design.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+## Chapter outline
+
+Course chapters and content need to be strongly aligned with learning outcome/goal.
+Here is a draft outline to guide development.
+
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - What makes this course different
+
```
Comment (cwickham) [inline r-rse/course-design.md:56]: I'm a big fan of the "Whole Game" chapter in R Packages: it removes some of the magic/fear around packages by walking through making one from scratch.  I think it's an activity that's doable in less than an hour and gives students a quick win.  Is this something we could do up front?  Or is this what you imagine "Setting up an R package" looks like?

**Comment by lwjohnst86** [comment]
Thanks for the comments @cwickham! I've added them to the file, closing this, and opening a PR in the new repo.

**Pull request closed without merging**